app/hotels/src/allHotels/AllHotels.js

```diff
@@ -77,78 +83,69 @@ export class AllHotels extends React.Component<PropsWithContext> {
-      <Layout>
-        <ScrollView
-          bounces={false}
-          contentContainerStyle={styles.scrollViewContainer}
-          onScroll={Keyboard.dismiss}
-        >
-          <SearchForm
-            onChange={this.props.setSearch}
-            search={this.props.search}
-            location={this.props.location}
-            openLocationPicker={this.openLocationPicker}
-            openGuestsModal={this.props.openGuestsModal}
-          />
-          <FilterStripe
-            filter={this.props.filter}
-            onChange={this.props.setFilter}
-            currency={this.props.currency}
-          />
-          {this.props.cityId || this.props.coordinates ? (
-            <AllHotelsSearch
-              coordinates={this.props.coordinates}
-              openSingleHotel={this.props.openSingleHotel}
-              cityId={this.props.cityId}
-              currency={this.props.currency}
+  render = () => {
+    const coordinates = this.getCoordinates();
+    const canRenderAllHotelsSearch = this.props.cityId || coordinates;
+
+    return (
+      <AppStateChange
+        states={['active']}
+        onStateChange={this.validateCheckinDate}
+      >
+        <Layout>
+          <ScrollView
+            bounces={false}
+            contentContainerStyle={styles.scrollViewContainer}
+            onScroll={Keyboard.dismiss}
+          >
+            <SearchForm
+              onChange={this.props.setSearch}
+              search={this.props.search}
+              location={this.props.location}
+              openLocationPicker={this.openLocationPicker}
+              openGuestsModal={this.props.openGuestsModal}
             />
-          ) : (
-            <PublicApiRenderer
-              query={graphql`
-                query AllHotels_cityLookup_Query($prefix: String!) {
-                  city: hotelCities(prefix: $prefix, first: 1) {
-                    edges {
-                      node {
-                        id
-                      }
-                    }
-                  }
-                }
-              `}
-              render={this.renderAllHotelsSearchPublicRenderer}
-              variables={{
-                prefix: this.props.location,
-              }}
+            <FilterStripe
+              filter={this.props.filter}
+              onChange={this.props.setFilter}
+              currency={this.props.currency}
             />
-          )}
-        </ScrollView>
-      </Layout>
-    </AppStateChange>
-  );
+            {canRenderAllHotelsSearch ? (
+              <AllHotelsSearch
+                coordinates={coordinates}
+                openSingleHotel={this.props.openSingleHotel}
+                cityId={this.props.cityId}
+                currency={this.props.currency}
+              />
+            ) : (
+              <View style={styles.searchForCityContainer}>
+                {!this.props.canGetUserLocation ? (
+                  <Translation id="hotels_search.all_hotels.please_search" />
```

Comment:
So this will allow us to avoid the API renderer in order to fetch initial locations, right? I just want to confirm the idea.

Reply:
yes